src/py/mat3ra/made/tools/build/interface.py

```diff
@@ -1,268 +1,280 @@
+            new_name = f"{updated_material.name}, Strain (mean_abs_strain={strain:.3f}%)"
+            updated_material.name = new_name
+        return material
 
-class InterfaceDataHolder(object):
+    def _finalize(
+        self, materials: List[Material], configuration: InterfaceBuilder._ConfigurationType
+    ) -> List[Material]:
+        return [self._update_material_name(material, configuration) for material in materials]
+
+
+class ZSLStrainMatchingInterfaceBuilder(StrainMatchingInterfaceBuilder):
     """
-    A class to hold data for interfaces generated by pymatgen.
-    Structures are stored in a dictionary with the termination as the key.
-    Example data structure:
-        {
-            "('C_P6/mmm_2', 'Si_R-3m_1')": [
-                { ...interface for ('C_P6/mmm_2', 'Si_R-3m_1') at index 0...},
-                { ...interface for ('C_P6/mmm_2', 'Si_R-3m_1') at index 1...},
-                ...
-            ],
-            "<termination at index 1>": [
-                { ...interface for 'termination at index 1' at index 0...},
-                { ...interface for 'termination at index 1' at index 1...},
-                ...
-            ]
-        }
+    Creates matching interface between substrate and film using the ZSL algorithm.
     """
 
-    def __init__(self, entries: Union[InterfacesType, None] = None) -> None:
-        if entries is None:
-            entries = []
-        self.data: InterfacesDataType = {}
-        self.terminations: List[TerminationType] = []
-        self.add_data_entries(entries)
+    _BuildParametersType: type(  # type: ignore
+        ZSLStrainMatchingInterfaceBuilderParameters
+    ) = ZSLStrainMatchingInterfaceBuilderParameters  # type: ignore
+    _GeneratedItemType: PymatgenInterface = PymatgenInterface  # type: ignore
+
+    def _generate(self, configuration: InterfaceConfiguration) -> List[PymatgenInterface]:
+        generator = ZSLGenerator(**self.build_parameters.strain_matching_parameters.dict())
+        builder = CoherentInterfaceBuilder(
+            substrate_structure=to_pymatgen(configuration.substrate_configuration.bulk),
+            film_structure=to_pymatgen(configuration.film_configuration.bulk),
+            substrate_miller=configuration.substrate_configuration.miller_indices,
+            film_miller=configuration.film_configuration.miller_indices,
+            zslgen=generator,
+        )
+
+        # TODO: REMOVE when found the correct solution
+        # Workaround: compare provided and generated terminations to find complete match,
+        # if full match between terminations isn't found, get terminations with first part the same, before `_`
+
+        generated_terminations = builder.terminations
+        provided_termination_pair = configuration.termination_pair
+        hotfix_termination_pair = configuration.termination_pair
+
+        for termination_pair in generated_terminations:
+            if termination_pair == provided_termination_pair:
```

Comment:
This is unnecessary, you should only check the case where termination_parin is not in generated_terminations